docs/lucene.adoc

```diff
@@ -23,6 +23,7 @@ In the above configuration, the index backend is named `search`. Replace `search
 * *Full-Text*: Supports all `Text` predicates to search for text properties that matches a given word, prefix or regular expression.
 * *Geo*: Supports `Geo` predicates to search for geo properties that are intersecting, within, or contained in a given query geometry. Supports points, lines and polygons for indexing. Supports circles and boxes for querying point properties and all shapes for querying non-point properties. Note that JTS is required when using line and polygon shapes (see <<search-predicates#geoshape,Geoshape documentation>> for more information).
 * *Numeric Range*: Supports all numeric comparisons in `Compare`.
+* *Collections*: Supports indexing SET and LIST cardinality properties.
```

Comment:
except Geoshape

Reply:
I'll update the doc asap.

I'm not sure that #670 and #764 can be easily separated because, 
fixing #764 required a complete rewrite of the way mutation were handled and the way stuff was stored... and this is intertwined with Set, List support (if you want to implement this later on, you better design storage and mutation in a way that will allow implementing them later on, so you have to think about that before the rewrite).

BUT if you want to have the bug fix and don't want to have set/list support in 0.2, you can just comment/disable
those line 
        .supportsCardinality(Cardinality.LIST)
        .supportsCardinality(Cardinality.SET)
in LuceneIndex
And you can enable them in master

Set and list support for Geoshape would require a complete redesign of how geoshapes are stored and queried (and I'm not willing/ready to do that now)